unitTests/ui/ResultLayoutSelectorTest.ts

```diff
@@ -0,0 +1,44 @@
+import { ResultLayoutSelector } from '../../src/ui/ResultLayoutSelector/ResultLayoutSelector';
+import { ValidLayout } from '../../src/ui/ResultLayoutSelector/ValidLayout';
+import * as Mock from '../MockEnvironment';
+import { $$ } from '../../src/utils/Dom';
+import { ResultLayoutEvents, IResultLayoutPopulateArgs } from '../../src/events/ResultLayoutEvents';
+import { InitializationEvents } from '../../src/events/InitializationEvents';
+
+export function ResultLayoutSelectorTest() {
+  describe('ResultLayoutSelector', () => {
+    let test: Mock.IBasicComponentSetup<ResultLayoutSelector>;
+
+    function handleLayoutsPopulation(rootElement: HTMLElement) {
+      $$(rootElement).on(ResultLayoutEvents.populateResultLayout, (e, args: IResultLayoutPopulateArgs) => {
+        args.layouts.push(...ResultLayoutSelector.validLayouts);
+      });
+    }
+
+    function triggerRootEvent(event: string) {
+      $$(test.env.root).trigger(event);
+    }
+
+    function buildResultLayoutSelector(activeLayout: ValidLayout) {
+      test = Mock.advancedComponentSetup<ResultLayoutSelector>(ResultLayoutSelector, <Mock.AdvancedComponentSetupOptions>{
+        modifyBuilder: builder => {
+          handleLayoutsPopulation(builder.root);
+          (builder.queryStateModel.get as jasmine.Spy).and.returnValue(activeLayout);
+          return builder;
+        }
+      });
+      triggerRootEvent(InitializationEvents.afterComponentsInitialization);
+      triggerRootEvent(InitializationEvents.afterInitialization);
+    }
+
+    it('if the list layout is selected, should give the pressed state to the list button', () => {
+      buildResultLayoutSelector('list');
+      expect(test.cmp['currentActiveLayouts']['list'].button.el.getAttribute('aria-pressed')).toEqual(true.toString());
```

Comment:
Line is a bit long, could you extract parts of it into variables with explicit names?

Reply:
Oh I see `test.cmp['currentActiveLayouts']['list'].button.el.getAttribute('aria-pressed')` is repeated, you could extract it to a method

Reply:
Good point, I'll extract the `getButton(layout)` method.